x-vector/src/X/Data/Vector/Stream/Group.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module X.Data.Vector.Stream.Group (
+    embed
+  , concatWith
+  , chunksOf
+  ) where
+
+import           Control.Monad.Base (MonadBase(..))
+import           Control.Monad.Primitive (PrimMonad(..))
+
+import           Data.Vector.Fusion.Stream.Monadic (Stream(..), Step(..))
+import qualified Data.Vector.Fusion.Stream.Monadic as Stream
+import qualified Data.Vector.Generic as Generic
+
+import           P
+
+import qualified Prelude as Savage
+
+import qualified X.Data.Vector.Grow as Grow
+
+
+embed :: Monad m => m (Stream m a) -> Stream m a
+embed mstream =
+  Stream.concatMapM (const mstream) (Stream.singleton ())
+{-# INLINE [1] embed #-}
+
+concatWith :: Monad m => Stream m a -> (a -> Stream m b) -> Stream m b
+concatWith =
+  flip Stream.concatMap
+{-# INLINE [1] concatWith #-}
+
+chunksOf :: (PrimMonad b, MonadBase b m, Generic.Vector v a) => Int -> Stream m a -> Stream m (v a)
+chunksOf !n (Stream step t) =
+  if n <= 0 then
+    Savage.error "X.Data.Vector.Stream.Group.chunksOf: chunk size must be greater than zero"
```

Comment:
Not sure what to do about this, I could alternatively yield chunks of 1, but perhaps that indicates a bug?